src/main/java/redis/clients/jedis/HostAndPort.java

```diff
@@ -47,6 +47,38 @@ public String toString() {
     return host + ":" + port;
   }
 
+  /**
+   * Parses String.
+   * String must be in ( host + ":" + port ) format.
+   * Port is optional
+   * @param from String to parse
+   * @return array of host and port strings
+     */
+  public static String[] extractParts(String from){
```

Comment:
Parsing IPv6 addresses will always find a port, no matter whether the port is specified or not.